src/components/footer.tsx

```diff
@@ -0,0 +1,86 @@
+import { Todo } from '../types/Todo';
+import classNames from 'classnames';
+import * as todosService from '../api/todos';
+
+type Props = {
+  onFilter: (filter: string) => void;
+  onError: () => void;
+  onTodos: () => void;
+  todos: Todo[];
+  filter: string;
+};
+
+export const Footer: React.FC<Props> = ({
+  onFilter,
+  onError,
+  onTodos,
+  todos,
+  filter,
+}) => {
+  return (
+    <footer
+      className={classNames('todoapp__footer', { hidden: todos.length === 0 })}
+      data-cy="Footer"
+      style={{ display: todos.length === 0 ? 'none' : '' }}
+    >
+      <span className="todo-count" data-cy="TodosCounter">
+        {todos.filter(todo => !todo.completed).length} items left
+      </span>
+
+      {/* Active link should have the 'selected' class */}
+      <nav className="filter" data-cy="Filter">
+        <a
+          href="#/"
+          className={classNames('filter__link', {
+            selected: filter === 'all',
+          })}
+          data-cy="FilterLinkAll"
+          onClick={() => onFilter('all')}
+        >
+          All
+        </a>
+
+        <a
+          href="#/active"
+          className={classNames('filter__link', {
+            selected: filter === 'active',
+          })}
+          data-cy="FilterLinkActive"
+          onClick={() => onFilter('active')}
+        >
+          Active
+        </a>
+
+        <a
+          href="#/completed"
+          className={classNames('filter__link', {
+            selected: filter === 'completed',
```

Comment:
Use `Object.values(your created enum)` and render these options with `map()` method